pkg/condition/conditions_test.go

```diff
@@ -178,6 +154,92 @@ func TestCondition_IsWindowsIPAMEnabled(t *testing.T) {
 	}
 }
 
+func TestCondition_GetPodDataStoreSyncStatus(t *testing.T) {
+	tests := []struct {
+		name            string
+		testBlockedTime int
+		// buffer time is used to consider other execution time during the process
+		testBufferTime int
+		workers        int
+	}{
+		{
+			name:            "worker doesn't wait long enough",
+			testBlockedTime: 5,
+			testBufferTime:  -1,
+			workers:         1,
+		},
+		{
+			name:            "worker waits with a buffer time",
+			testBlockedTime: 5,
+			testBufferTime:  1,
+			workers:         1,
+		},
+		{
+			name:            "three workers check on the flag",
+			testBlockedTime: 5,
+			testBufferTime:  1,
+			workers:         3,
+		},
+		{
+			name:            "large number workers check on the flag",
+			testBlockedTime: 5,
+			testBufferTime:  1,
+			workers:         30, // we want to use an unusual large workers to test routines conflict
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockK8s := mock_k8s.NewMockK8sWrapper(ctrl)
+			conditions := NewControllerConditions(zap.New(), mockK8s)
+			start := time.Now()
+
+			// one routine is keeping the flag as false for 5s and then updates it to true
+			go func() {
+				waitFor := 0
+				for waitFor < test.testBlockedTime {
+					time.Sleep(time.Second * 1)
+					conditions.SetPodDataStoreSyncStatus(false)
+					waitFor++
+				}
+				conditions.SetPodDataStoreSyncStatus(true)
+			}()
+
+			var wg sync.WaitGroup
+			wg.Add(test.workers)
+
+			for i := 0; i < test.workers; i++ {
```

Comment:
I think we need some more comments explaining this test case, as I am confused on what the workers are doing here. It seems like they are just consuming the sync status, so I am missing how this helps cover the multiple reader/writer scenarios

Reply:
Please check the response above.